### PR TITLE
WW-601 | banner notifications for RWE page header experiment

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/WW-601

@Wikia/west-wing

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/a88c20219800f15be84f8cc28c2317da4ef53487/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/RWEPageHeader/styles/index.scss [hunk 1]
```diff
@@ -25,3 +27,30 @@
 .WikiHeader {
 	display: none;
 }
+
+.WikiaTopAds {
```
Comment (sebastianmarzjan) [inline extensions/wikia/RWEPageHeader/styles/index.scss]: Why are we changing this margin?

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 2]
```diff
@@ -19,5 +19,44 @@ require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracke
...
+
+		window.BannerNotification.prototype.hide = function() {
+			this.onHide();
+
+			var $globalNav = $('#globalNavigation');
+			var $notificationsWrapper = $('.banner-notifications-wrapper');
+
+			$notificationsWrapper.css('top', ($globalNav.position().top + 55) + 'px');
+
+			if ($globalNav.position().top == 0) {
```
Comment (WikiaTech) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js]: MAJOR Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/44e11331a3bd182974f3fdf6b009949ba6da12fe/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/3b28c1ef516e777735b6721006a437fa50b2b6ac/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/d4fbc82b26f4e8edd696aede760aa0a7db4ba359/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/8cbff5f3b9e681470147ec3595e656dc23771bc3/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/d4fbc82b26f4e8edd696aede760aa0a7db4ba359/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracker) {
+require(['wikia.window', 'jquery', 'wikia.tracker', 'wikia.onScroll'], function (window, $, tracker, onScroll) {
```
Comment (WikiaTech) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js]: MAJOR Remove the unused function parameter "onScroll". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/74b9c9f413953da37a4d4ecd817a5e15de6c011d/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 4]
```diff
@@ -19,5 +19,44 @@ require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracke
...
+
+
+		var $wikiaTopAds = $('.WikiaTopAds');
+		var $globalNav = $('.wds-global-navigation-wrapper');
+
+
+		window.BannerNotification.prototype.onShow = window.BannerNotification.prototype.show;
+		window.BannerNotification.prototype.onHide = window.BannerNotification.prototype.hide;
+
+		function setNotificationPosition() {
```
Comment (WikiaTech) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js]: MAJOR Remove unused function 'setNotificationPosition'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 5]
```diff
@@ -19,5 +19,44 @@ require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracke
...
+			}
+		}
+
+		window.BannerNotification.prototype.show = function() {
+			this.onShow();
+			var $notificationsWrapper = $('.banner-notifications-wrapper');
+			$notificationsWrapper.detach();
+
+			$notificationsWrapper.insertAfter($globalNav);
+			// setNotificationPosition();
```
Comment (WikiaTech) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 6]
```diff
@@ -19,5 +19,44 @@ require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracke
...
+
+			return this;
+		};
+
+		window.BannerNotification.prototype.hide = function() {
+			this.onHide();
+			// setNotificationPosition();
```
Comment (WikiaTech) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 5 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/6796fd18f48ba7f43c3722873fb8a7bdc470fe0c/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 7]
```diff
@@ -19,5 +19,28 @@ require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracke
 		$('.rwe-page-header-nav__element-dropdown > .rwe-page-header-nav__link').on('click', function (e) {
 			e.preventDefault();
 		});
+
+
+
+		// var $wikiaTopAds = $('.WikiaTopAds');
```
Comment (WikiaTech) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/6cbbcd9d8758b41bb9b5ef86d1f71bbf6fe00a74/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/09518c74a8b57b4c7ef7bb938bf747b9aa4596b9/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [rwe-page-header.js#L13](https://github.com/Wikia/app/blob/2c55800be8a45632f6ca1b646d2d910e294c2204/extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js#L13): Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 8]
```diff
@@ -10,16 +10,36 @@ require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracke
 	};
 
 	$(function () {
+		var $globalNav = $('.wds-global-navigation-wrapper'),
+			$notificationsWrapper = $('.banner-notifications-wrapper');
+
 		$('.rwe-page-header-nav__link, .rwe-page-header-wordmark_wrapper').on('click', function (e) {
```
Comment (WikiaTech) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js:16]: MAJOR Remove the unused function parameter "e". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Review by lukaszkonieczny — APPROVED** [review]

File: extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js [hunk 9]
```diff
@@ -10,16 +10,36 @@ require(['wikia.window', 'jquery', 'wikia.tracker'], function (window, $, tracke
...
 			});
 		});
 
+
 		$('.rwe-page-header-nav__element-dropdown > .rwe-page-header-nav__link').on('click', function (e) {
 			e.preventDefault();
 		});
 
+		$notificationsWrapper.detach();
+		$notificationsWrapper.insertAfter($globalNav);
```
Comment (sebastianmarzjan) [inline extensions/wikia/RWEPageHeader/scripts/rwe-page-header.js:28]: Nitpick: duplicated logic inside of this closure

**Review by sebastianmarzjan — APPROVED** [review]

**Pull request merged**